src/watts/plugin_pyarc.py

```diff
@@ -0,0 +1,186 @@
+# SPDX-FileCopyrightText: 2022 UChicago Argonne, LLC
+# SPDX-License-Identifier: MIT
+
+from contextlib import redirect_stdout, redirect_stderr
+from datetime import datetime
+from functools import lru_cache
+from pathlib import Path
+import shutil, os
+import time
+from typing import Callable, Mapping, List
+
+import h5py
+import sys
+
+from .fileutils import PathLike, tee_stdout, tee_stderr
+from .parameters import Parameters
+from .plugin import TemplatePlugin
+from .results import Results
+
+
+class ResultsPyARC(Results):
+    """OpenMC simulation results
+
+    Parameters
+    ----------
+    params
+        Parameters used to generate inputs
+    time
+        Time at which workflow was run
+    inputs
+        List of input files
+    outputs
+        List of output files
+
+    Attributes
+    ----------
+
+    """
+
+    def __init__(self, params: Parameters, time: datetime,
+                 inputs: List[Path], outputs: List[Path], user_object):
+        super().__init__('PyARC', params, time, inputs, outputs, user_object)
+        self.results_data = self._save_PyARC(user_object)
+
+    @property
+    def stdout(self) -> str:
+        return (self.base_path / "PyARC_log.txt").read_text()
+
+    def _save_PyARC(self, user_object) -> dict:
+        """Return PyARC results in a dictionary
+
+        Returns
+        -------
+        Results from PyARC results
+
+        """
+        results_data = {}
+        results_data["mcc3"] = user_object.results_keff_mcc3
+        results_data["dif3d"] = user_object.results_keff_dif3d
+        results_data["proteus_nodal"] = user_object.results_keff_nodal
+        results_data["rzmflx"] = user_object.results_keff_rzmflx
+        results_data["persent_pert"] = user_object.results_kpert_persent
+        results_data["persent_sens"] = user_object.results_kuq_persent
+        results_data["gamsor"] = user_object.results_power_gamsor
+        results_data["dassh"] = user_object.results_dassh
+        # TODO - this is minimum information that can be brought back easily. 
+        # More should be returned but will require work on PyARC.
+        return results_data
+
+    def save(self, filename: PathLike):
+        """Save results to an HDF5 file
+
+        Parameters
+        ----------
+        filename
+            File to save results to
+        """
+        with h5py.File(filename, 'w') as h5file:
+            super()._save(h5file)
+
+    @classmethod
+    def _from_hdf5(cls, obj: h5py.Group):
+        """Load results from an HDF5 file
+
+        Parameters
+        ----------
+        obj
+            HDF5 group to load results from
+        """
+        time, parameters, inputs, outputs = Results._load(obj)
+        return cls(parameters, time, inputs, outputs)
+
+class PluginPyARC(TemplatePlugin):
```

Comment:
Per [PEP8](https://www.python.org/dev/peps/pep-0008/#blank-lines), two blank lines before module-level functions/classes
```suggestion


class PluginPyARC(TemplatePlugin):
```